tt_metal/impl/program/program.cpp

```diff
@@ -516,6 +516,11 @@ void Program::add_semaphore(const CoreRangeSet & crs, uint32_t address, uint32_t
     semaphores_.emplace_back(Semaphore( crs, address, init_value, core_type));
 }
 
+void Program::add_config_buffer(std::shared_ptr<Buffer> config_buffer) {
+    this->invalidate_compile();
```

Comment:
Why does adding the config buffer invalidate compile? Program is compiled with no knowledge of this.

Reply:
Hmm, if the op config (buffer) is different for the same program config, we want to distinguish it. But thinking about it, currently we do not have different config for same program config. So I can remove this.